src/value.rs

```diff
@@ -215,6 +220,86 @@ impl UserData for Color3uint8Value {
     }
 }
 
+#[derive(Debug, Clone, Copy)]
+pub struct Vector3Value(Vector3);
+
+impl fmt::Display for Vector3Value {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}, {}, {}", self.0.x, self.0.y, self.0.z)
+    }
+}
+
+impl Vector3Value {
+    pub fn new(value: Vector3) -> Self {
+        Self(value)
+    }
+
+    fn meta_index<'lua>(
+        &self,
+        context: Context<'lua>,
+        key: &str,
+    ) -> rlua::Result<rlua::Value<'lua>> {
+        match key {
+            "X" => self.0.x.to_lua(context),
+            "Y" => self.0.y.to_lua(context),
+            "Z" => self.0.z.to_lua(context),
+            _ => Err(rlua::Error::external(format!(
+                "'{}' is not a valid member of Vector3",
+                key
+            ))),
+        }
+    }
+}
+
+impl ops::Add for &Vector3Value {
```

Comment:
You should be able to impl these ops for `Vector3Value` instead of `&Vector3Value` since the type is `Copy`. You might need to dereference in a couple places, but I think it ends up a little clearer!